File: 3rd Semester/Advanced Programming Methods/labs/10th lab/decorator/DelayTaskRunner.cs
namespace Seminar_10.decorator;
using System.Threading.Tasks;

public class DelayTaskRunner: AbstractTaskRunner
{
    public DelayTaskRunner(ITaskRunner taskRunner) : base(taskRunner){
    }

    public override void ExecuteAll()
    {
        while (HasTask())
        {
            ExecuteOneTask();
            Console.WriteLine("Waiting for 3 seconds...");
            try
            {
                Thread.Sleep(3000); 
            }
            catch (ThreadInterruptedException e)
            {
                Console.WriteLine(e);
            }
            Console.WriteLine($"Task executed at: {DateTime.Now.ToString("HH:mm:ss")}");
        }
        Console.WriteLine("All tasks have been executed.");
        }
}

File: 3rd Semester/Advanced Programming Methods/labs/2nd lab/decorator/DelayTaskRunner.java
package sem1_2.decorator;

public class DelayTaskRunner extends AbstractTaskRunner {

    public DelayTaskRunner(TaskRunner taskRunner) {
        super(taskRunner);
    }

    @Override
    public void executeAll(){
        while(hasTask()){
            executeOneTask();
            try{
                Thread.sleep(3000);
            }
            catch(InterruptedException e){
                e.printStackTrace();
            }
        }
    }
}
